app-backend/src/user/ServiceManager.js

```diff
@@ -452,12 +452,25 @@ class ServiceManager {
                     && (customDomain.length < 80)
                     && /^[a-z0-9\-\.]+$/.test(customDomain)
                     && (customDomain.indexOf('..') < 0)
-                    && (customDomain.indexOf(rootDomain) < 0 || (customDomain === rootDomain) || ((customDomain.indexOf(rootDomain) + rootDomain.length) !== customDomain.length));
+                    && (customDomain.indexOf("."+rootDomain) < 0 || (customDomain === rootDomain) || ((customDomain.indexOf(rootDomain) + rootDomain.length) !== customDomain.length));
 
-                if (!isAllowed) {
+                if (!customDomain) {
                     throw ApiStatusCodes.createError(ApiStatusCodes.STATUS_ERROR_BAD_NAME,
                         'Domain name is not accepted. Please use alphanumerical domains such as myapp.google123.ca');
                 }
+                if (customDomain.length > 80) {
+                    throw ApiStatusCodes.createError(ApiStatusCodes.STATUS_ERROR_BAD_NAME,
+                        'Domain name is not accepted. Please use alphanumerical domains less than 80 characters in length.');
+                }
+                if (customDomain.indexOf('..') > 0) {
+                    throw ApiStatusCodes.createError(ApiStatusCodes.STATUS_ERROR_BAD_NAME,
+                        'Domain name is not accepted. You cannot have two consecutive periods ".." inside a domain name. Please use alphanumerical domains such as myapp.google123.ca');
+                }
+                if (!(customDomain.indexOf("."+rootDomain) < 0 || (customDomain === rootDomain) || ((customDomain.indexOf(rootDomain) + rootDomain.length) !== customDomain.length))) {
```

Comment:
`rootDomain` in the last condition also needs to be replaced with `"."+rootDomain`.
Better to create a temp variable like `let dotRootDomain = "." + rootDomain` in order to avoid repeating adding the dot prefix.